### PR TITLE
Add edit text example

### PR DESCRIPTION
**Pull request by lord**
It's pretty slow and the code is kind of crappy, but I think it's good enough for review. Having a text editing example is I think important to make sure our text editing APIs don't degrade.

Chinese text input no longer shows the composition window; still have to look into why that is, but perhaps I can fix that in another PR.

### CODE REVIEW

**Comment by xStrom** [comment]
Clippy errors were resolved with #77 so you can rebase for CI success.

**Comment by lord** [comment]
(rebase complete, this should be ready for review)

File: examples/edit_text.rs [hunk 1]
```diff
@@ -0,0 +1,509 @@
...
+const HEIGHT: usize = 1536;
+const FONT_SIZE: f32 = 36.0;
+const TEXT_X: f64 = 100.0;
+const TEXT_Y: f64 = 100.0;
+
+fn main() {
+    let app = Application::new().unwrap();
+    let mut window_builder = glazier::WindowBuilder::new(app.clone());
+    window_builder.resizable(true);
+    window_builder.set_size((WIDTH as f64 / 2., HEIGHT as f64 / 2.).into());
```
Comment (jneem) [inline examples/edit_text.rs:39]: Why divide by 2?

File: examples/edit_text.rs [hunk 2]
```diff
@@ -0,0 +1,509 @@
...
+
+impl Default for ParleyBrush {
+    fn default() -> ParleyBrush {
+        ParleyBrush(Brush::Solid(Color::rgb8(0, 0, 0)))
+    }
+}
+
+impl PartialEq<ParleyBrush> for ParleyBrush {
+    fn eq(&self, _other: &ParleyBrush) -> bool {
+        true // FIXME
```
Comment (jneem) [inline examples/edit_text.rs:294]: What's this?

**Review by jneem — APPROVED** [review]
Looks good! I had a couple of silly little questions, but my main question is: what's the deal with the `#[non_exhaustive]` on `HitTestPoint`? I assume it was originally put there for a reason, but I'm missing a bunch of context here...

**Comment by lord** [comment]
Thanks for reviewing! the `non_exhaustive` was something I put there in my last PR; I'm pretty sure I blindly copied it from another struct without thinking.

**Comment by jneem** [comment]
Ah, gotcha. I noticed that it's present in piet, so I figured that's where it came from.

File: examples/edit_text.rs (hunk 2 above)
Comment (lord) [reply examples/edit_text.rs:294]: Copied over from the `shello.rs` example :/

File: examples/edit_text.rs (hunk 1 above)
Comment (lord) [reply examples/edit_text.rs:39]: This is another copy from `shello.rs` :/

**Comment by lord** [comment]
Oop, you're totally right! I must have copied it from there. Looks like in piet this was possible because there was a constructor on the type, so I've gone ahead and added back in the `non_exhaustive` annotation and added a constructor.

File: examples/edit_text.rs (hunk 2 above)
Comment (dfrg) [reply examples/edit_text.rs:294]: `peniko::Brush` is `PartialEq` and has been for a while so we should be able to just derive here now. I think this was a limitation in old piet-gpu code. We may need to update some revs for this to be visible.

**Pull request merged**